src/OpenTelemetry.Instrumentation.StartupHook/StartupHook.cs

```diff
@@ -65,4 +78,44 @@ private static string GetLoaderAssemblyLocation()
             throw;
         }
     }
+
+    private static string GetApplicationName()
+    {
+        try
+        {
+            return AppDomain.CurrentDomain.FriendlyName;
+        }
+        catch (Exception ex)
+        {
+            StartupHookEventSource.Log.Error($"Error getting AppDomain.CurrentDomain.FriendlyName: {ex}");
```

Comment:
Under normal usage of a start up hook I don't think we can hit this exception although we can hit it in code that are calling the startup hook directly. Are we not risking hiding improver usage with this catch here?

Reply:
Related: what do we get for an app started with `dotnet run` that still has to build the app? Any difference between the friendly names in Windows and Linux? Perhaps, it is worth a comment on the code or link to the docs.